conductor_api/src/conductor/base.rs

```diff
@@ -113,28 +119,52 @@ impl Conductor {
             instances: HashMap::new(),
             interface_threads: HashMap::new(),
             static_servers: HashMap::new(),
+            broadcasters: Arc::new(RwLock::new(HashMap::new())),
             config,
             dna_loader: Arc::new(Box::new(Self::load_dna)),
             ui_dir_copier: Arc::new(Box::new(Self::copy_ui_dir)),
-            signal_tx: None,
+            signal_tx: Initable::Uninit,
             logger: DebugLogger::new(rules),
             p2p_config: None,
             network_child_process: None,
         }
     }
 
-    pub fn with_signal_channel(mut self, signal_tx: SyncSender<Signal>) -> Self {
-        if !self.instances.is_empty() {
-            panic!("Cannot set a signal channel after having run load_config()");
+    fn setup_signals(&mut self, maybe_signal_tx: Option<SignalSender>) {
+        if let Some(signal_tx) = maybe_signal_tx {
+            self.signal_tx = Initable::Init(signal_tx);
+        } else {
+            let (signal_tx, signal_rx) = signal_channel();
+            self.signal_tx = Initable::Init(signal_tx);
+            self.start_signal_broadcast(signal_rx);
         }
-        self.signal_tx = Some(signal_tx);
-        self
     }
 
     pub fn config(&self) -> Configuration {
         self.config.clone()
     }
 
+    pub fn start_signal_broadcast(&mut self, signal_rx: SignalReceiver) -> thread::JoinHandle<()> {
```

Comment:
this function I think needs some rusdoc comments because it's a `pub` function and feels like an important place where people will look.